src/Tooltip/index.js

```diff
@@ -0,0 +1,122 @@
+import { createPortal } from 'react-dom'
+import React, { Component, createRef } from 'react'
+import propTypes from 'prop-types'
+import css from 'styled-jsx/css'
+import cx from 'classnames'
+
+import { reactRef } from '../prop-validators/reactRef'
+import { BackgroundCover } from './BackgroundCover'
+import { Content } from './Content'
+import {
+    arePositionsEqual,
+    getPosition,
+    getScrollAndClientOffset,
+    propPosition,
+} from './helpers'
+
+/**
+ * The Tooltip component is a content container that behaves like a context menu
+ * container. It can be used to create multi level context menus that won't be
+ * displayed off-screen by wrapping each level with the Tooltip component.
+ */
+class Tooltip extends Component {
+    ref = createRef()
+    state = {
+        position: {},
+        adjustment: {},
+    }
+
+    componentDidMount() {
+        this.updatePosition()
+        window.addEventListener('resize', this.updatePosition)
+    }
+
+    updatePosition = () => {
+        if (this.props.open && this.ref.current) {
+            const { anchorRef, side, spacing } = this.props
+
+            const [position, adjustment] = getPosition({
+                pop: this.ref.current,
+                anchor: anchorRef.current,
+                side,
+                spacing,
+            })
+
+            if (!arePositionsEqual(position, this.state.position)) {
+                this.setState({ position, adjustment })
+            }
+        }
+    }
+
+    render() {
+        if (!this.props.open) return null
+
+        const { side, children, onClose, withArrow } = this.props
+        const { position, adjustment } = this.state
+
+        return createPortal(
+            <div>
+                <BackgroundCover onClick={onClose} />
```

Comment:
I'm not sure I think that a Tooltip should have a background cover. That feels extremely invasive UX-wise.

Ping @cooper-joe.

Reply:
Correct, a tooltip should not have a background cover. 
(However, is this PR not for the popover? Either way, neither tooltip nor popover should use a Cover.)

Reply:
It's just an invisible cover to catch the click outside of the tooltip.

Reply:
It's invisible, but it still blocks the UI. It's possible to handle the "click outside" without a div covering 100% of the screen.